packages/babel-types/src/definitions/experimental.js

```diff
@@ -89,6 +89,28 @@ defineType("OptionalMemberExpression", {
   },
 });
 
+defineType("PipelineTopicExpression", {
+  builder: ["expression"],
+  visitor: ["expression"],
+  fields: {
+    expression: {
+      validate: assertNodeType("Expression"),
+    },
+  },
+});
+
+defineType("PipelineBareFunction", {
+  builder: ["callee"],
+  visitor: ["callee"],
+  fields: {
+    callee: {
+      validate: assertNodeType("Expression"),
+    },
+  },
+});
+
+defineType("PipelinePrimaryTopicReference", {});
```

Comment:
I think that this should still be an expression, since it can safely* moved around.

* It's safe only if it is in the correct context, like it would be for any other identifier reference.